[dynamo] Support itertools.groupby

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114192

Summary: for https://github.com/pytorch/pytorch/issues/108698

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

desertfire committed:
[dynamo] Support itertools.groupby

Summary: for https://github.com/pytorch/pytorch/issues/108698

[ghstack-poisoned]

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel commented on torch/_dynamo/variables/misc.py:
```diff
@@ -848,6 +848,66 @@ def call_function(
+
+            result = []
+            last_pair = None
+            for item in seq:
```
You could likely replace this loop with a call it `itertools.groupby`

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire commented:
@pytorchbot rebase

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

desertfire commented:
@pytorchbot rebase

desertfire committed:
Update on "[dynamo] Support itertools.groupby"


Summary: for https://github.com/pytorch/pytorch/issues/108698

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]
